src/scripts/utils/ParatiiLib.js

```diff
@@ -29,3 +29,28 @@ if (process.env.NODE_ENV === 'test') {
 }
 
 export const paratii: ParatiiLib = window.paratii
+
+export async function setupKeystore () {
+  let defaultPassword = 'password'
+  let mnemonic
+  let walletKey = 'keystore-anon'
+  let mnemonicKey = 'mnemonic-anon'
+  let existingWallet = localStorage.getItem(walletKey)
+  let existingWalletIsValid
+  if (existingWallet) {
+    try {
+      console.log('Found existing wallet')
+      paratii.eth.wallet.decrypt(JSON.parse(existingWallet), defaultPassword)
+    } catch (err) {
+      console.log('Existing wallet is not valid')
+      existingWalletIsValid = false
+    }
+  }
+  if (!existingWallet || existingWalletIsValid === false) {
+    console.log('Creating a new wallet')
+    mnemonic = await paratii.eth.wallet.newMnemonic()
+    localStorage.setItem(walletKey, JSON.stringify(paratii.eth.wallet.encrypt(defaultPassword)))
+    localStorage.setItem(mnemonicKey, mnemonic)
+  }
+  // console.log(`Your account is: ${}`)
```

Comment:
If we go with an approach like the one I propose in [this comment](https://github.com/Paratii-Video/paratii-portal/pull/53#issuecomment-357456716), then this logic would instead go into a redux thunk action in a `WalletActions.js` module

Reply:
It does not sound wrong :-) - but I am not sure I am the right person to judge :-)

Some of this state (e.g. the euser's ethereum address) is kept in the `paratii` object as well. Most of this state will probably be set just once in the lifetime of the app.  So we'd have 3 places to store the address:
   - paratii.config.address
   - redux
   - localstorage

Reply:
Actually the situation is mre complex. In localStorage, we store only the encrypted version of the wallet, not the wallet itself. (And similarly for the mnemonic: it is now stored in celar tes, but should be encrypted as well)

At this point, it is not clear to me what we would gain by keeping these values in te redux state.

Reply:
My understanding is that whether or not we have some of these data (e.g. address, encrypted wallet) and what their values are will control at least some logic in our application, yes? If so, then I think we definitely need it in the redux store. With redux and its connected container components, changes to the store/state will automatically re-render components based on store changes, if these components' properties coming from the store have changed. If we have data living outside of our store (e.g. in `localStorage`) then it is up to use to manually respond to these changes and re-render when necessary. That would introduce an additional layer of complexity to our app, I think, and we should avoid it.

One of the main tenants of Redux is single-source of truth/unidirectional data flow, and if we go against that pattern I think we could end up w/ a mess on our hands

Reply:
Let's discuss this by voice :-)